smack-extensions/src/main/java/org/jivesoftware/smackx/softwareinfo/SoftwareInfoListener.java

```diff
@@ -0,0 +1,26 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.softwareinfo;
+
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smackx.softwareinfo.form.SoftwareInfoForm;
+
+import org.jxmpp.jid.BareJid;
+
+public interface SoftwareInfoListener {
```

Comment:
The listener is now dead code and should be removed.

Besides the order of the parameters should have been changed: Parameters should be ordered by importance, the most important parameter coming first. If this listener was still in use, then users would first want to know which entity changed their information, and secondly the actual changed information, and the carrier message is the most unimportant information as it will not be used in most cases.